mmcls/models/backbones/edgenext.py

```diff
@@ -0,0 +1,401 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import math
+from itertools import chain
+from typing import Sequence
+
+import torch
+import torch.nn as nn
+from mmcv.cnn.bricks import DropPath, build_activation_layer, build_norm_layer
+from mmcv.runner.base_module import BaseModule, ModuleList, Sequential
+
+from mmcls.models.backbones.convnext import ConvNeXtBlock
+from ..builder import BACKBONES
+from ..utils import ChannelMultiheadAttention, PositionEncodingFourier
+from .base_backbone import BaseBackbone
+
+
+class SDTAEncoder(BaseModule):
+    """A PyTorch implementation of split depth-wise transpose attention (SDTA)
+    encoder.
+
+    Inspiration from
+    https://github.com/mmaaz60/EdgeNeXt
+
+    Args:
+        in_channel (int): Number of input channels.
+        drop_path_rate (float): Stochastic depth dropout rate.
+            Defaults to 0.
+        layer_scale_init_value (float): Initial value of layer scale.
+            Defaults to 1e-6.
+        mlp_ratio (int): Number of channels ratio in the MLP.
+            Defaults to 4.
+        use_pos_emb (bool): Whether to use position encoding.
+            Defaults to True.
+        num_heads (int): Number of heads in the multihead attention.
+            Defaults to 8.
+        qkv_bias (bool): Whether to use bias in the multihead attention.
+            Defaults to True.
+        attn_drop (float): Dropout rate of the attention.
+            Defaults to 0.
+        proj_drop (float): Dropout rate of the projection.
+            Defaults to 0.
+        layer_scale_init_value (float): Initial value of layer scale.
+            Defaults to 1e-6.
+        norm_cfg (dict): Dictionary to construct normalization layer.
+            Defaults to ``dict(type='LN')``.
+        act_cfg (dict): Dictionary to construct activation layer.
+            Defaults to ``dict(type='GELU')``.
+        scales (int): Number of scales. Default to 1.
+    """
+
+    def __init__(self,
+                 in_channel,
+                 drop_path_rate=0.,
+                 layer_scale_init_value=1e-6,
+                 mlp_ratio=4,
+                 use_pos_emb=True,
+                 num_heads=8,
+                 qkv_bias=True,
+                 attn_drop=0.,
+                 proj_drop=0.,
+                 norm_cfg=dict(type='LN'),
+                 act_cfg=dict(type='GELU'),
+                 scales=1):
+        super().__init__()
```

Comment:
The module inherits `BaseModule` should accept `init_cfg` and pass it to the superclass.